### PR TITLE
Initial PMIC interfacing, changes in logging function[added log type]…

### PR DESCRIPTION
**Pull request by krzysztofkrzeslak**
…, SSID validation, added ip in network info

### CODE REVIEW

**Review by mdelain — CHANGES_REQUESTED** [review]
`addPowerStatus()` seems to freeze the board :(
Tested on LTE-M & GSM boards

> It seems that `Wire.Begin()` is too far away from the requests made inside `LiveObjectsBase::addPowerStatus()`.
> I made a simple test by disabling the `batteryBegin()` function (by doing so, I disabled the first `Wire.Begin()`), and adding the `Wire.Begin()` directly before the I2C requests in `readRegister()`
```c++
byte readRegister(byte address) {
    Wire.begin();   // added here
    Wire.beginTransmission(PMIC_ADDRESS);
    Wire.write(address);

    if (Wire.endTransmission(true) != 0) {
      return -1;
    }

    if (Wire.requestFrom(PMIC_ADDRESS, 1, true) != 1) {
      return -1;
    }

    return Wire.read();
}
```
> It seems to work smoothly now.

> The behavior is strange on the MKR VIDOR, it says that a battery is connected even if not (!). I searched a bit but didn't find any information for the moment. It's the same PMIC though.

**Review by mdelain — CHANGES_REQUESTED** [review]
One more left in `Utils.cpp` : https://github.com/DatavenueLiveObjects/LiveObjects_SDK_for_Arduino/blob/50b3d4e2d4a590efc82b6ea91abf990030874a36/src/Utils.cpp#L26

File: src/LiveObjects.cpp [hunk 1]
```diff
@@ -300,16 +299,15 @@ void LiveObjectsBase::sendData(const String customPayload) {
 }
 
 void LiveObjectsBase::publishMessage(const String& topic, JsonDocument& payload) {
-  outputDebug("Publishing message on topic");
-  outputDebug(topic);
+  outputDebug(INFO,"Publishing message on topic", topic);
```
Comment (mdelain) [inline src/LiveObjects.cpp]: There is a missing space after "topic", leading to something like
_[INFO] Publishing message on **topicdev/data**_

**Review by mdelain — CHANGES_REQUESTED** [review]
You need to add a specific hack for MKR1000, because it doesn't support TLS.
So choosing TLS should trigger a stop in the code.

**Review by mdelain — CHANGES_REQUESTED** [review]
You could add support of ARDUINO_SAMD_MKRVIDOR4000 (using WiFiNINA library). However as PMIC is not working properly for now (see in PMIC comments), let's not define it as compatible with the feature.

**Comment by krzysztofkrzeslak** [comment]
1. Seem that on the MKR1500 issue was that the address of charging IC was swapped in comparison to MKRWIFI1010, should be fixed now, can you check if this is also true for MKR Vidor?
2. Serial.println() in utils is not a mistake as output debug can be used only in the LiveObjects class.

**Comment by krzysztofkrzeslak** [comment]
No more change requests, requested ones already done, merging to master.

**Pull request merged**